src/charm.py

```diff
@@ -202,6 +217,37 @@ def _generate_vault_config_file(self) -> None:
                 source=content,
             )
 
+    def _set_peer_relation_node_api_address(self) -> None:
+        """Set the unit address in the peer relation."""
+        peer_relation = self.model.get_relation(PEER_RELATION_NAME)
+        if not peer_relation:
+            raise RuntimeError("Peer relation not created")
```

Comment:
Do we really want to raise in this case? I have seen the peer relation get created a bit later in the lifecycle of a charm, and in that case, the creation of the relation would trigger a new event.

Reply:
I was missing a validation in `_configure` to make sure the peer relation is created. We should not get inside of this method here if the peer relation is not created.

Reply:
The problem with using `assert` in production code is that it can be disabled at runtime. In this case it would still error out with an exception, but it is probably better to keep it like this.

Reply:
Done